### PR TITLE
Add FAQ advising against underscores in hostnames.

### PR DESCRIPTION
**Pull request by davet2001**
## Description:

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2079

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: guides/faq.rst [hunk 1]
```diff
@@ -343,6 +343,14 @@ with a lot of work).
 
 Always back up all your files!
 
+
+Why shouldn't I use underscores in my device name?
+--------------------------------------------------
+
+The top level ``name:`` field in your .yaml file defines the node name(/hostname) on the local network.  According to `RFC1912 <https://datatracker.ietf.org/doc/html/rfc1912>`_, underscore characters (``_``) in hostnames are not valid.  In reality some local DNS/DHCP setups will be ok with underscores and some will not.  If connecting via a static IP address, there will probably be no issues.  In some cases, initial setup using an underscore works, but then the connection fails when Home Assistant restarts.  Recommendation: use hyphen (``-``) instead of underscore if you can.
```
Comment (glmnet) [inline guides/faq.rst]: >  but then the connection fails when Home Assistant restarts.

could this be more like: "but then the connection **might** fail when Home Assistant restarts."

File: guides/faq.rst [hunk 2]
```diff
@@ -343,6 +343,14 @@ with a lot of work).
 
 Always back up all your files!
 
+
+Why shouldn't I use underscores in my device name?
+--------------------------------------------------
+
+The top level ``name:`` field in your .yaml file defines the node name(/hostname) on the local network.  According to `RFC1912 <https://datatracker.ietf.org/doc/html/rfc1912>`_, underscore characters (``_``) in hostnames are not valid.  In reality some local DNS/DHCP setups will be ok with underscores and some will not.  If connecting via a static IP address, there will probably be no issues.  In some cases, initial setup using an underscore works, but then the connection fails when Home Assistant restarts.  Recommendation: use hyphen (``-``) instead of underscore if you can.  
+
+Important: follow these `instructions </components/esphome.html#changing-esphome-node-name>`_ to use the ``use_address`` parameter when renaming a live device, as the connection to an existing device will only work with the old name until the name change is complete.
```
Comment (glmnet) [inline guides/faq.rst:352]:
```suggestion
Follow these `instructions </components/esphome.html#changing-esphome-node-name>`_ to change the underscores in the name to hyphens.
```

No need to explain why here, it's explained in the instructions.

**Review by glmnet — COMMENTED** [review]
I see no network gear is talk about, I mean, an advise that underscores can work with your current network router but might stop working if you upgrade those, makes sense?

File: guides/faq.rst (hunk 2 above)
Comment (davet2001) [reply guides/faq.rst:352]: I think it is better to give some idea why here.  Many will assume they can just rename it.

File: guides/faq.rst (hunk 1 above)
Comment (davet2001) [reply guides/faq.rst]: Agreed, done.

**Comment by davet2001** [comment]
> I see no network gear is talk about, I mean, an advise that underscores can work with your current network router but might stop working if you upgrade those, makes sense?

Agreed. Updated.

File: guides/faq.rst [hunk 3]
```diff
@@ -343,6 +343,14 @@ with a lot of work).
 
 Always back up all your files!
 
+
```
Comment (OttoWinter) [inline guides/faq.rst:346]:
```suggestion
.. _faq-device_name_underscores:
```

So we can link to it better

Comment (OttoWinter) [reply guides/faq.rst:346]: Nvm, apparently the link is not generated with that.

**Pull request merged**